File: week11/20221104/크레인-인형뽑기-게임/javascript/Solution.js
function solution(board, moves) {
    var answer = 0;
  var arr = []
  
    for(var i = 0; i < moves.length; i++){
        for(var j = 0; j < board.length;j++){
            if(board[j][moves[i]-1] !== 0){
                arr.push(board[j][moves[i]-1])
                board[j][moves[i]-1] = 0
                break;
            }
        }
      
        if (arr.length >=2){
            if (arr[arr.length-1] === arr[arr.length-2]){
                answer +=2
                arr.splice(arr.length-2,2)

            }
        } 
    }
    return answer;
}

File: week11/20221104/크레인-인형뽑기-게임/java/src/Solution.java
import java.util.ArrayList;
import java.util.List;

public class Solution {
    public int solution(int[][] board, int[] moves) {
        return count(pop(board, moves));
    }

    public List<Integer> pop(int[][] board, int[] moves) {
        List<Integer> numbers = new ArrayList<>();

        for (int i = 0; i < moves.length; i++) {
            for (int j = 0; j < board.length; j++) {
                if (board[j][moves[i] - 1] != 0) {
                    numbers.add(board[j][moves[i] - 1]);
                    board[j][moves[i] - 1] = 0;
                    break;
                }
            }
        }

        return numbers;
    }

    public int count(List<Integer> basket) {
        int count = 0;

        for (int k = 0; k < basket.size(); k++) {
            if (k != (basket.size() - 1)) {
                if (basket.get(k) == basket.get(k + 1)) {
                    basket.remove(k);
                    basket.remove(k);
                    count += 2;
                    k = -1;
                }
            }
        }

        return count;
    }
}

